include/seqan3/search/fm_index/fm_index_cursor.hpp

```diff
@@ -32,6 +32,45 @@ namespace seqan3
  * \{
  */
 
+//!\brief The underlying suffix array intervals.
+struct suffix_array_interval
+{
+    //!\brief The begin position of the interval ("left boundary").
+    size_t begin_position;
+    //!\brief The exclusive end position of the interval ("right boundary").
+    size_t end_position;
+
+    //!\brief Convert to tuple.
+    operator std::tuple<size_t &, size_t &>()
+    {
+        return {begin_position, end_position};
+    }
+
+    /*!\name Comparison operators
+     * \{
+     */
+    /*!\brief Test for equality.
+     * \param[in] lhs A `seqan3::suffix_array_interval`.
+     * \param[in] rhs `seqan3::suffix_array_interval` to compare to.
+     * \returns `true` if equal, `false` otherwise.
+     */
+    friend bool operator==(suffix_array_interval const & lhs, suffix_array_interval const & rhs) noexcept
+    {
+        return std::tie(lhs.begin_position, lhs.end_position) == std::tie(rhs.begin_position, rhs.end_position);
```

Comment:
```suggestion
        return lhs.begin_position == rhs.begin_position && lhs.end_position == rhs.end_position;
```

name me old fashioned, but why not like this?